llmfoundry/utils/builders.py

```diff
@@ -115,6 +115,10 @@ def build_optimizer(cfg: DictConfig, model: torch.nn.Module):
                                   timeout=cfg.timeout,
                                   lr_penalty=cfg.lr_penalty,
                                   min_scale=cfg.min_scale)
+    elif cfg.name.lower() == 'decoupled_lionw_8b':
+        # str() cast is just for pyright
+        kwargs = {str(k): v for k, v in cfg.items() if k != 'name'}
```

Comment:
if you want to make all optimizer inits use `**kwargs`, that would help people avoid wrong args
and you can probably `cfg.pop('name')` at the top of the builder fn

Reply:
Can I do this in a separate PR? I admittedly wasn't super disciplined about this but it would be nice to not risk breaking something unrelated in this one.